java/arcs/core/storage/StorageProxy.kt

```diff
@@ -26,12 +26,15 @@ import kotlinx.coroutines.launch
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 
+// Visible for testing
+enum class ProxyState { INIT, AWAITING_SYNC, SYNC, DESYNC }
```

Comment:
Re. moving into the SP class - doesn't that mean test code will need to use `StorageProxy.ProxyState.XXX`? I was trying to avoid that verbosity, but on reflection it's not really an issue. Will update.

Good point on the KDoc - will also update.

Reply:
If you want to avoid seeing a bunch of `StorageProxy.ProsyState.Blah` stuff in tests, you can add an `import ...StorageProxy.ProxyState` to the test's file and keep the same syntax as if it were top level

Reply:
I did consider using a `typealias` (I guess the direct import would be better?) but figured the verbosity is harmless - it didn't cause any extra line breaks - and makes any given use quite clear.

I also tried marking the state with `internal` visibility, but the test code still isn't allowed to access that, despite being declared in the same package. Is that known behaviour?

Reply:
Actually I will use the import trick.

Tangential question: if I mark the state enum and getter method as `internal`, the test code doesn't have access despite being declared in the same package. Is that expected?

Reply:
It's a bug in Bazel's kotlin rules, they don't handle kotlin's notion of "friends" correctly (compared to Gradle, for example)